File: 01_basics/01_08_all_divisors/alldivisors.js
const print_all_divisors = (n) => {
    process.stdout.write(`All divisors of ${n} are: `)
    for (let index = 1; index <= Math.sqrt(n); index++) {
        if (n % index === 0) {
            process.stdout.write(String(index) + " ");
            if (Math.floor(n / index) !== index) {
                process.stdout.write(String(Math.floor(n / index)) + " ");
            }
        }
    }
}

const num = 36
print_all_divisors(num);

File: 01_basics/01_08_all_divisors/AllDivisors.java
public class AllDivisors {
    public static void main(String[] args) {
        int num = 36;
        printAllDivisors(num);
    }

    private static void printAllDivisors(int n) {
        System.out.print("All divisors of " + n + " are: ");
        for (int i = 1; i < Math.sqrt(n) + 1; i++) {
            if (n % i == 0) {
                System.out.print(i + " ");
                if (i != n / i) {
                    System.out.print(n / i + " ");
                }
            }
        }
    }
}
